IOSExercises/Views/ProfilesTableViewController.m

```diff
@@ -61,39 +69,105 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
     ProfileViewController *profileViewController = [[ProfileViewController alloc] initWithNibName:@"ProfileViewController"
                                                                                            bundle:[NSBundle mainBundle]];
-    [self.persons objectAtIndex:indexPath.item];
-    [self.navigationController pushViewController:profileViewController
-                                         animated:YES];
+
+    if (indexPath.item < self.persons.count) {
```

Comment:
When adding safety checks, better to use guard pattern where the invalid case gets blocked before the code for valid cases are implemented. eg.

```
if (<index is invalid>) {
  <Trigger assertion and log>
  <Handle, eg return nil>
}

<Handle valid case>
```